### PR TITLE
Dependency caching blog WIP

### PR DESCRIPTION
**Pull request by komalali**
### Proposed changes

Blog post for dependency caching

### CODE REVIEW

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+---
+title: "Announcing dependency caching for Pulumi Deployments"
+date: 2024-11-11
```
Comment (arunkumar611) [inline content/blog/announcing-dependency-caching-deployments/index.md:3]: Updated timestamp for this to show above the CosmosDB post

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+1.	Check for an existing cache of your dependencies.
+2.	If a cache is available and relevant to your project, Pulumi will reuse those dependencies, bypassing the need to install them from scratch.
+3.	If dependencies have changed, Pulumi will update the cache to reflect these changes, ensuring it remains accurate for subsequent deployments.
+
+The end result? Faster, more efficient deployments that get your infrastructure changes live more quickly.
+
+## How to enable dependency caching
+
+Dependency caching can be enabled through deployment settings, however you choose to configure them.
```
Comment (arunkumar611) [inline content/blog/announcing-dependency-caching-deployments/index.md:33]:
```suggestion
Dependency caching can be enabled in numerous ways: 
```

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+authors:
+- komal-ali
+- iaroslav-titov
+tags:
+- features
+- pulumi-deployments
+- caching
+---
+
+We’re excited to announce dependency caching for Pulumi Deployments! This new feature is designed to significantly speed up deployment times by caching dependencies between runs, ensuring faster and more efficient workflows for every Pulumi user.
```
Comment (arunkumar611) [inline content/blog/announcing-dependency-caching-deployments/index.md:15]: In the best case, can it be 75% faster? If yes, would be good to call out.

**Comment by arunkumar611** [comment]
the blog post is also missing links. At the minimum, we should link to
- Pulumi Deployments Docs
- Pulumi Cloud webpage 

If we are requesting for feedback at the end, I'd link the cloud requests repo link as well.

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+When using Pulumi Deployments, installing dependencies can be a time-consuming part of a deployment. With dependency caching, Pulumi can reuse previously installed dependencies, allowing your deployments to skip redundant installation steps and focus on the task at hand. This optimization is especially helpful for frequent deployments, reducing both time and compute resources.
+
+<!--more-->
+
+## How dependency caching works
+
+Pulumi Deployments now automatically caches dependencies for popular package managers, such as npm, pip, and go modules. When a deployment is triggered, Pulumi will:
+
+1.	Check for an existing cache of your dependencies.
+2.	If a cache is available and relevant to your project, Pulumi will reuse those dependencies, bypassing the need to install them from scratch.
```
Comment (IaroslavTitov) [inline content/blog/announcing-dependency-caching-deployments/index.md:26]: `available and relevant to your project` sounds a bit too wordy, maybe `If a cache already exists` or similar?

Also maybe change `If dependencies have changed` to `Otherwise, or if dependencies have changed` + to make sure people understand how cache gets created/not just updated, but this one might be hard to reword the whole sentence

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 2 above)
Comment (IaroslavTitov) [reply content/blog/announcing-dependency-caching-deployments/index.md:33]: I kinda like the current version better

**Review by IaroslavTitov — APPROVED** [review]
A couple comments, overall lgtm!

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+<!--more-->
+
+## How dependency caching works
+
+Pulumi Deployments now automatically caches dependencies for popular package managers, such as npm, pip, and go modules. When a deployment is triggered, Pulumi will:
+
+1.	Check for an existing cache of your dependencies.
+2.	If a cache already exists, Pulumi will reuse those dependencies, bypassing the need to install them from scratch.
+3.	If a cache does not exist, or if dependencies have changed, Pulumi will create a new cache to be used for subsequent deployments.
```
Comment (cleverguy25) [inline content/blog/announcing-dependency-caching-deployments/index.md:27]:
```suggestion
3.	If a cache does not exist, or if dependencies have changed, Pulumi will manage the cache to be used for subsequent deployments.
```

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+The end result? Faster, more efficient deployments that get your infrastructure changes live more quickly.
```
Comment (cleverguy25) [inline content/blog/announcing-dependency-caching-deployments/index.md:29]: Maybe it is just me but I read the other form of "live" than intended and had to read it twice.

Comment (komalali) [reply content/blog/announcing-dependency-caching-deployments/index.md:29]: "deployed more quickly"?

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    },
+    cacheOptions: {
+        enable: true,
+    }
+});
+```
+
+If your project uses a supported package manager, Pulumi will automatically detect and manage dependency caching for you.
+
+For more information on how dependency caching works with Pulumi Cloud Deployments, check out the [FAQ](/docs/pulumi-cloud/deployments/faq/#dependency-caching) on dependency caching.
```
Comment (cleverguy25) [inline content/blog/announcing-dependency-caching-deployments/index.md:82]: Do we have a PR to update that FAQ page?

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+            repoDir: "pulumi-programs/simple-resource"
+        }
+    },
+    cacheOptions: {
+        enable: true,
+    }
+});
+```
+
+If your project uses a supported package manager, Pulumi will automatically detect and manage dependency caching for you.
```
Comment (cleverguy25) [inline content/blog/announcing-dependency-caching-deployments/index.md:80]: Where do we list "supported package managers"?

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 7 above)
Comment (komalali) [reply content/blog/announcing-dependency-caching-deployments/index.md:82]: Is there information that you think should be there that isn't currently? https://www.pulumi.com/docs/pulumi-cloud/deployments/faq/#dependency-caching

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 8 above)
Comment (komalali) [reply content/blog/announcing-dependency-caching-deployments/index.md:80]: In the FAQ but this could maybe be clearer.

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 7 above)
Comment (cleverguy25) [reply content/blog/announcing-dependency-caching-deployments/index.md:82]: Didn't know that was live already, disregard!

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 6 above)
Comment (cleverguy25) [reply content/blog/announcing-dependency-caching-deployments/index.md]: Maybe "The end result? Faster, more efficient deployments to push out infrastructure changes quickly."

File: content/blog/announcing-dependency-caching-deployments/index.md (hunk 8 above)
Comment (cleverguy25) [reply content/blog/announcing-dependency-caching-deployments/index.md:80]: Yeah, with the link to the FAQ in the next paragraph, I may be left wondering where the approved list is.

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+<!--more-->
+
+## How dependency caching works
+
+Pulumi Deployments now automatically caches dependencies for popular package managers, such as npm, pip, and go modules. When a deployment is triggered, Pulumi will:
+
+1.	Check for an existing cache of your dependencies.
+2.	If a cache already exists, Pulumi will reuse those dependencies, bypassing the need to install them from scratch.
+3.	If a cache does not exist, or if dependencies have changed, Pulumi will manage the cache to be used for subsequent deployments.
```
Comment (nyobe) [inline content/blog/announcing-dependency-caching-deployments/index.md:27]:
```suggestion
3.	If a cache does not exist, or if dependencies have changed, Pulumi will install fresh dependences and create an updated cache to be used for subsequent deployments.
```

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+
+1. Open the Pulumi Console and navigate to a stack's deployment settings (i.e. Stack page > Settings > Deploy).
+2. Under the "Dependency caching" section, toggle the switch to enable dependency caching for your stack.
+3. Click "Save deployment configuration" to apply the changes.
+
+![Dependency caching in the Pulumi Console](enable-caching-console.png)
+
+### Using the REST API
+
+You can also enable dependency caching using the Pulumi REST API. See the [API documentation](/docs/pulumi-cloud/deployments/api/#cache-options) for more information.
```
Comment (nyobe) [inline content/blog/announcing-dependency-caching-deployments/index.md:45]: `"You can also ..."` reads to me as the final item in a list of options, but then its followed up by PSP.  Maybe move the REST API item to the end?

File: content/blog/announcing-dependency-caching-deployments/index.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+    },
+    sourceContext: {
+        git: {
+            repoUrl: "https://github.com/pulumi/deploy-demos.git",
+            branch: "refs/heads/main",
+            repoDir: "pulumi-programs/simple-resource"
+        }
+    },
+    cacheOptions: {
+        enable: true,
```
Comment (nyobe) [inline content/blog/announcing-dependency-caching-deployments/index.md:75]: I think putting a comment on the line that actually enables the feature would be nice, eg
```suggestion
        enable: true, // <--
```

**Review by nyobe — APPROVED** [review]
As a reader, one question I'm left with after reading this post is when would you _not_ want to enable dependency caching?

**Review by cleverguy25 — APPROVED** [review]
